src/algorithms.rs

```diff
@@ -2,6 +2,7 @@ use num_bigint::traits::ModInverse;
 use num_bigint::{BigUint, RandPrime};
 use num_traits::{FromPrimitive, One, Zero};
 use rand::Rng;
+use libm::F64Ext;
```

Comment:
Could you update to libm@0.2, I did that in num-bigint-dig, unfortunately that doesn't have the extension anymore, so you need to do some manual wrapping of things.